src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/NotificationRuleController.scala

```diff
@@ -0,0 +1,63 @@
+/*
```

Comment:
Missing
private class DagInstanceStatusesDeserializer extends JsonDeserializer[DagInstanceStatus] {
    override def deserialize(p: JsonParser, ctxt: DeserializationContext): DagInstanceStatus = {
      val node = p.getCodec.readTree[JsonNode](p)
      val value = node.get("name").textValue()
      DagInstanceStatuses.statuses.find(_.name == value).getOrElse(throw new Exception("Failed to find enum value"))
    }
  }

in Application.scala
Spring needs that for json deserialization